Src/MoneyFox.Ui/Views/Settings/SettingsViewModel.cs

```diff
@@ -31,4 +40,33 @@ public CurrencyViewModel SelectedCurrency
     }
 
     public IReadOnlyList<CurrencyViewModel> AvailableCurrencies { get; }
+
+    public AccountLiteViewModel? SelectedAccount
+    {
+        get => selectedAccount ??= AvailableAccounts?.FirstOrDefault(x => x.Name == (string.IsNullOrEmpty(settingsFacade.DefaultAccount) ? string.Empty : settingsFacade.DefaultAccount));
+
+        set
+        {
+            SetProperty(field: ref selectedAccount, newValue: value);
+            settingsFacade.DefaultAccount = selectedAccount != null ? selectedAccount.Name : string.Empty;
+            OnPropertyChanged();
+        }
+    }
+
+    public void LoadAccounts()
+    {
+        availableAccounts = mediator.Send(new GetAccountsQuery()).Result.Select(a => new AccountLiteViewModel(a.Name)).ToList();
+    }
+
+    public IEnumerable<AccountLiteViewModel>? AvailableAccounts
+    {
+        get
+        {
+            if(availableAccounts == null)
+            {
+                LoadAccounts();
+            }
```

Comment:
Is this needed still? With the loadAccounts this should never be the case, right?

Reply:
It is required as I mentioned here https://github.com/MoneyFox/MoneyFox/pull/2996#discussion_r1279821335

Reply:
@NPadrutt The issue is that SelectedAccount is dependent on AvailableAccounts and SelectedAccount data is fetched by UI before 'OnAppearing' is called.

To resolve this I have called LoadAccouts () and updated SelectedAccount when the constructor is called. But this change resulted in the failure of the unit test ServiceCollectionTest. Ther error message is below 
`SQLite Error 1: 'no such table: Accounts'.)
---- Microsoft.Data.Sqlite.SqliteException : SQLite Error 1: 'no such table: Accounts'.

I am struck here and need your help.`

Reply:
The way I solved this in other cases is by calling an Initialize Method in the `OnAppearing` overload in the page. For example in the EditAccountPage is an example of that. With that you don't have to make db calls in the constructor but only when you actually navigating to the page and need the data.

For that to work you would have to change the AvailableAccounts to:

```
    private List<AccountLiteViewModel> availableAccounts = new();
    public List<AccountLiteViewModel> AvailableAccounts
    {
        get => availableAccounts;
        set => SetProperty(field: ref availableAccounts, newValue: value);
    }
```

`SetProperty` also calls  `OnPropertyChanged` so the binding is updated with that.
A bit unfortunate is, that with this solution the default selection is empty. But you could for example set `SelectedAccount` to the first account when it's empty.

Reply:
btw, I think you can simplify this:
```
 AvailableAccounts = accounts == null ? new List<AccountLiteViewModel>() : accounts.Select(a => new AccountLiteViewModel(a.Id, a.Name)).ToList();
```

To

```
        AvailableAccounts = accounts.Select(a => new AccountLiteViewModel(Id: a.Id, Name: a.Name)).ToList();
```

`GetAccountsQuery` returns a non-null List. So worst case is that the List is empty in which case select is not executed and you just get an empty AccountLiteViewModel list :)